### PR TITLE
Adding the first 4 tasks in text to video about file

### PR DESCRIPTION
**Pull request by snknitin**
This is a PR that adds description to the text-to-video task page. it includes  

Task variants 
Use cases 
Most Used Dataset for the Task 
Most Used model for this task

### CODE REVIEW

**Comment by snknitin** [comment]
@merveenoyan  @osanseviero  This is related to the recent [PR](https://github.com/huggingface/hub-docs/issues/789). Please guide me on the steps for merge and if there is any changes required. I don't see most common datasets or models on other tasks, so i included it in the about.md . I wasn't sure where to add them

File: tasks/src/text-to-video/about.md
Comment (merveenoyan) [inline tasks/src/text-to-video/about.md:3]: ``
```suggestion
Text-to-video models can be used in any application that requires generating temporally and spatially consistent sequence of images from text.  

```

File: tasks/src/text-to-video/about.md
Comment (merveenoyan) [inline tasks/src/text-to-video/about.md:8]:
```suggestion
Text-to-video models can be used to create short-form video content from a provided text script. These models can be used to create engaging and informative marketing videos. For example, a company could use a text-to-video model to create a video that explains how their product works or that shows how their product can solve a customer's problem.

```

File: tasks/src/text-to-video/about.md
Comment (merveenoyan) [inline tasks/src/text-to-video/about.md:17]:
```suggestion
Text-to-video models can be used to generate videos that show people how to stay safe during a hurricane or how to find food and water after a natural disaster.

```

File: tasks/src/text-to-video/about.md
Comment (merveenoyan) [inline tasks/src/text-to-video/about.md:21]:
```suggestion
```

I think this is a bit redundant as we mention how it can be used for educational purposes above.

File: tasks/src/text-to-video/about.md
Comment (merveenoyan) [inline tasks/src/text-to-video/about.md:19]:
```suggestion
```

File: tasks/src/text-to-video/about.md
Comment (merveenoyan) [inline tasks/src/text-to-video/about.md:25]:
```suggestion
```

File: tasks/src/text-to-video/about.md
Comment (merveenoyan) [inline tasks/src/text-to-video/about.md:27]:
```suggestion
Text-to-video models can be used to create an AI newscaster to deliver daily news, or for a film-maker to create a short film or a music video.

```

Comment (merveenoyan) [reply tasks/src/text-to-video/about.md:27]: Maybe mention that for this it should be combined with a TTS model :')

File: tasks/src/text-to-video/about.md
Comment (merveenoyan) [inline tasks/src/text-to-video/about.md:30]:
```suggestion
### Video Summarization

```

File: tasks/src/text-to-video/about.md
Comment (merveenoyan) [inline tasks/src/text-to-video/about.md:32]:
```suggestion
T2V models can be used to summarize long videos into shorter, more concise videos. This can be useful for people who want to quickly get the gist of a video without having to watch the entire thing.

```

Can you elaborate on how this can be done? I feel like this task would be video-to-video and not text-to-video.

File: tasks/src/text-to-video/about.md
Comment (merveenoyan) [inline tasks/src/text-to-video/about.md:37]:
```suggestion
Text-to-video models can be used to create visual stories to communicate ideas and emotions. For example, the model could be used to create a video that tells the story of a refugee family, or a natural landscape that might be difficult to capture under normal circumstances.

```

File: tasks/src/text-to-video/about.md
Comment (merveenoyan) [inline tasks/src/text-to-video/about.md:42]:
```suggestion
Creating music visualization is a complex, time-consuming, and resource-intensive process. Users define start and end prompts of music to parameterize the visualization and these prompts are warped between and generated according to the beat of the music for audio-reactive video. Commonly used design patterns for improving generated videos are "transitions", which express shifts in color, time, subject, or style, and "holds", which encourage visual emphasis and consistency. 
```
Last sentence can be divided, and can you add where the text-to-video models come into play?

File: tasks/src/text-to-video/about.md
Comment (merveenoyan) [inline tasks/src/text-to-video/about.md:47]:
```suggestion
Generating text-editable and pose-controllable character videos have a rising demand for creating avatars. By utilizing easily obtained datasets (i.e., image pose pair and pose-free video) and a text-to-image model to obtain the pose-controllable character videos, this learns a zero-initialized convolutional encoder to encode the pose information and then fine-tune the motion of the network via a pose-free video dataset.

```
Can you simplify this for a broader audience? 😊 

>  this learns a zero-initialized convolutional encoder to encode the pose information and then finetune the motion of the network via a pose-free video dataset.

File: tasks/src/text-to-video/about.md
Comment (merveenoyan) [inline tasks/src/text-to-video/about.md:53]:
```suggestion
Text-to-video models have different variants based on inputs and outputs.

```

File: tasks/src/text-to-video/about.md
Comment (merveenoyan) [inline tasks/src/text-to-video/about.md:55]:
```suggestion
### Text-to-video Retrieval

```

File: tasks/src/text-to-video/about.md
Comment (merveenoyan) [inline tasks/src/text-to-video/about.md:60]:
```suggestion
### Text-to-video Editing

```

File: tasks/src/text-to-video/about.md
Comment (merveenoyan) [inline tasks/src/text-to-video/about.md:62]:
```suggestion
One text-to-video task is generating text-based video style and local attribute editing. Text-to-video editing models can make it easier to perform tasks like cropping, stabilization, color correction, resizing and audio editing consistently.

```

File: tasks/src/text-to-video/about.md
Comment (merveenoyan) [inline tasks/src/text-to-video/about.md:65]:
```suggestion
### Text-to-video Search

```

File: tasks/src/text-to-video/about.md
Comment (merveenoyan) [inline tasks/src/text-to-video/about.md:67]:
```suggestion
Text-to-video search is the task of retrieving videos that are relevant to a given text query. This can be challenging, as videos are a complex medium that can contain a lot of information. By using semantic analysis to extract the meaning of the text query, visual analysis to extract features from the videos, such as the objects and actions that are present in the video, and temporal analysis to categorize relationships between the objects and actions in the video, we can determine which videos are most likely to be relevant to the text query.

```

File: tasks/src/text-to-video/about.md
Comment (merveenoyan) [inline tasks/src/text-to-video/about.md:71]:
```suggestion
Text-driven video prediction is the task of generating a video sequence from a text description. Text description can be anything from a simple sentence to a detailed story. The goal of this task is to generate a video that is both visually realistic and semantically consistent with the text description.

```

How is this different from text-to-video generation? I think this is the main task here.

File: tasks/src/text-to-video/about.md
Comment (merveenoyan) [inline tasks/src/text-to-video/about.md:74]:
```suggestion
### Video Translation

```

File: tasks/src/text-to-video/about.md
Comment (merveenoyan) [inline tasks/src/text-to-video/about.md:76]:
```suggestion
Text-to-video translation models can translate videos from one language to another or allow to query the multilingual text-video model with non-English sentences. This can be useful for people who want to watch videos in a language that they don't understand, especially when multi-lingual captions are available for training.

```

Interesting, how is the input text and output video here? Can you explain?

File: tasks/src/text-to-video/about.md
Comment (merveenoyan) [inline tasks/src/text-to-video/about.md:89]: This section goes into data.ts, inside datasets :)

File: tasks/src/text-to-video/about.md
Comment (merveenoyan) [inline tasks/src/text-to-video/about.md:104]: Same with this one, it goes into data.ts :) you can find it in placeholder

**Review by merveenoyan — COMMENTED** [review]
Very neat PR! I left some comments :) thank you 🤗

File: tasks/src/text-to-video/about.md
Comment (snknitin) [reply tasks/src/text-to-video/about.md:38]: My thought process here is that if we have a text-to-video model generate something long for say 10 minutes, based on a very detailed and descriptive prompt, being able to cut the prompt size to only include the basics can now generate a video that explains the same content but in 2 minutes with only the main points from the updated summary prompt

File: tasks/src/text-to-video/about.md
Comment (snknitin) [reply tasks/src/text-to-video/about.md:83]: There is a latent embedding space where the text input in English is converted to a representation that translates to concepts used to generate videos. If we are training a model to generate video from an English caption, we could expand the dataset to have multiple language captions per video to form a larger vocabulary that has artificial code switching(replacing some random repeating words into other language vocab) and this helps borrows statistical properties and distributions from higher resource language like English , to some lower resource language where data or text annotation isn't available, let's say Turkish. Obviously all concepts cannot be expressed in English and there are words in other languages like French or German to describe situations or objects. Searching or prompting for a video in any language should be able to pull up or generate a similar video about the same concept

Comment (snknitin) [reply tasks/src/text-to-video/about.md:83]: There is a latent embedding space where the text input in English is converted to a representation that translates to concepts used to generate videos. If we are training a model to generate video from an English caption, we could expand the dataset to have multiple language captions per video to form a larger vocabulary that has artificial code switching(replacing some random repeating words into other language vocab) and this helps borrows statistical properties and distributions from higher resource language like English , to some lower resource language where data or text annotation isn't available, let's say Turkish. Obviously all concepts cannot be expressed in English and there are words in other languages like French or German to describe situations or objects. Searching or prompting for a video in any language should be able to pull up or generate a similar video about the same concept 

I was brainstorming use-cases . We can remove this if it doesn't align completely or sounds ambiguous.

File: tasks/src/text-to-video/about.md
Comment (snknitin) [reply tasks/src/text-to-video/about.md:111]: Could you link me to a placeholder

File: tasks/src/text-to-video/about.md
Comment (snknitin) [reply tasks/src/text-to-video/about.md:81]: I wrote it in a generic way but I wanted to frame this as a fine tuning of text to video models. Similar to how a stable diffusion or LLM can be trained on a global data and then finetuned to a niche sub data, text to video model can consume specific video series for finetuning and then predict what the next output of a text prompt would look like if it included connections from the fine tuning dataset. One application would be generating new episodes of a TV show

File: tasks/src/text-to-video/about.md
Comment (snknitin) [reply tasks/src/text-to-video/about.md:105]: I'll move it !

File: tasks/src/text-to-video/about.md
Comment (snknitin) [reply tasks/src/text-to-video/about.md:50]: Text, speech and music are all viewed as sequential encodings by the model. So text to video model is basically coming into play as an encoded representation of the music (between start to end prompt), and converted to video. I came across this idea for the use case from some youtube videos . We can remove it if not necessary

**Comment by snknitin** [comment]
@merveenoyan I made all the commits and moved the data and models to the other file and addressed all the comments. Let me know how to proceed on those 2-3 use cases and how to make PR from the draft

**Comment by snknitin** [comment]
@merveenoyan Can you let me know how to turn this draft PR into a full one? Please check the data.ts file as well

**Comment by Muskan149** [comment]
hi @snknitin, i can't find the text-to-video folder on hf's hub-docs. why is that so? sorry, i'm new to github contributions :)

**Comment by snknitin** [comment]
> hi @snknitin, i can't find the text-to-video folder on hf's hub-docs. why is that so? sorry, i'm new to github contributions :)

Hi @Muskan149 , I think this PR is currently under review. Once approved it should be visible, if I'm not wrong. But i think you can fork the hubdocs repo and create a folder by the same name and work on your part with the same file structure and add stuff in the about.md file.  The changes can be merged once you merge the Pull request. @merveenoyan Please correct me if i'm wrong. This is my first PR too

**Comment by tescao** [comment]
@Muskan149 @snknitin hi guys, same problem - I'm ready to contribute my part, but I think we'd better wait for the initial PR to be approved and merged so that we can pull that file and edit it. If each of us creates the file from scratch and PRs their own part, there's going to be a merge conflict as we'll be editing same lines in different ways, hence more work for everyone later on.

**Comment by snknitin** [comment]
> @Muskan149 @snknitin hi guys, same problem - I'm ready to contribute my part, but I think we'd better wait for the initial PR to be approved and merged so that we can pull that file and edit it. If each of us creates the file from scratch and PRs their own part, there's going to be a merge conflict as we'll be editing same lines in different ways, hence more work for everyone later on.

Yes, I've made the first PR. Waiting for review from @merveenoyan

File: tasks/src/text-to-video/data.ts
Comment (osanseviero) [inline tasks/src/text-to-video/data.ts:59]: The model IDs need to be valid hf.co/model repos

File: tasks/src/text-to-video/data.ts
Comment (snknitin) [reply tasks/src/text-to-video/data.ts]: I fixed it . Removed the ones that are not available and added both the dataset and models that exist on hf.  some of the id's are <username>/<dataset_name> . Is that fine?

File: tasks/src/text-to-video/data.ts
Comment (snknitin) [reply tasks/src/text-to-video/data.ts]: Modified it. This reduced the number of available models to just 2 though. Earlier I researched a couple of model that do text2video, but not all are available on the repo

File: tasks/src/text-to-video/data.ts
Comment (merveenoyan) [reply tasks/src/text-to-video/data.ts]: Yep it is!

File: tasks/src/text-to-video/about.md
Comment (merveenoyan) [inline tasks/src/text-to-video/about.md:30]: Can you remove some of the newlines? You can be inspired by the style in NLP task pages :)

File: tasks/src/text-to-video/data.ts
Comment (merveenoyan) [inline tasks/src/text-to-video/data.ts:6]: These are a bit too long, they should fit into single line, e.g. see Notes under datasets in this page: https://huggingface.co/tasks/image-segmentation

**Review by merveenoyan — COMMENTED** [review]
Can you also add a task summary? They're single lines appearing on top of every page. I gave very general feedbacks, happy to merge after they're addressed :)

File: tasks/src/text-to-video/data.ts
Comment (snknitin) [reply tasks/src/text-to-video/data.ts]: Done. Shortened them to a single sentence with brief description

File: tasks/src/text-to-video/about.md
Comment (snknitin) [reply tasks/src/text-to-video/about.md:26]: Done ! Both these changes have been completed

File: tasks/src/text-to-video/data.ts
Comment (merveenoyan) [inline tasks/src/text-to-video/data.ts:6]:
```suggestion
			description: "Microsoft Research Video to Text is a large-scale dataset for open domain video captioning",
```

File: tasks/src/text-to-video/data.ts
Comment (merveenoyan) [inline tasks/src/text-to-video/data.ts:10]:
```suggestion
			description: "UCF101 Human Actions dataset consists of 13,320 video clips from YouTube, with 101 classes.",
```

File: tasks/src/text-to-video/data.ts
Comment (merveenoyan) [inline tasks/src/text-to-video/data.ts:18]:
```suggestion
			description: "A dataset consisting of labeled video clips that show humans performing pre-defined basic actions with everyday objects.",
```

File: tasks/src/text-to-video/data.ts
Comment (merveenoyan) [inline tasks/src/text-to-video/data.ts:26]:
```suggestion
			description: "Large dataset of video clips with captions sourced from Youtube videos on complex tasks with step-by-step information.",
```

File: tasks/src/text-to-video/data.ts
Comment (merveenoyan) [inline tasks/src/text-to-video/data.ts:30]:
```suggestion
			description: "A dataset of short Flickr videos for the temporal localization of events with descriptions.",
```

File: tasks/src/text-to-video/data.ts
Comment (merveenoyan) [inline tasks/src/text-to-video/data.ts:46]:
```suggestion
			description: "A robust model for text-to-video generation.",
```

File: tasks/src/text-to-video/data.ts
Comment (merveenoyan) [inline tasks/src/text-to-video/data.ts:42]:
```suggestion
			description: "A strong model for video generation.",
```

File: tasks/src/text-to-video/data.ts
Comment (merveenoyan) [inline tasks/src/text-to-video/data.ts:52]: Can you please add summary? It's the description that appears on top of the page :)

File: tasks/src/text-to-video/data.ts
Comment (merveenoyan) [inline tasks/src/text-to-video/data.ts:14]:
```suggestion
			description: "A high-quality dataset for human action recognition in YouTube videos.",
```

**Review by merveenoyan — COMMENTED** [review]
I've mostly done nitpicking to even shorten descriptions, thanks a lot!

File: tasks/src/text-to-video/data.ts
Comment (merveenoyan) [inline tasks/src/text-to-video/data.ts:52]:
```suggestion
	summary:       "Text-to-video is the task of generating videos from text. ",
```

File: tasks/src/text-to-video/data.ts
Comment (merveenoyan) [inline tasks/src/text-to-video/data.ts:18]:
```suggestion
			description: "A dataset of video clips of humans performing pre-defined basic actions with everyday objects.",
```

**Review by merveenoyan — APPROVED** [review]
I think we can merge! great work! 
I will add task schema and couple of other things and we can ship. What is your hugging face username? @snknitin

**Comment by snknitin** [comment]
> I think we can merge! great work! I will add task schema and couple of other things and we can ship. What is your hugging face username? @snknitin

Awesome ! My hf username is [NikeZoldyck](https://huggingface.co/NikeZoldyck) . Thanks so much for this opportunity. This will be my first PR for 🤗

**Pull request merged**
